lib/linter/apply-disable-directives.js

```diff
@@ -16,6 +16,139 @@ function compareLocations(itemA, itemB) {
     return itemA.line - itemB.line || itemA.column - itemB.column;
 }
 
+/**
+ * Groups a set of directives into sub-arrays by their parent comment.
+ * @param {Directive[]} directives Unused directives to be removed.
+ * @returns {Directive[][]} Directives grouped by their parent comment.
+ */
+function groupByParentComment(directives) {
+    const groups = [];
+
+    if (!directives.length) {
+        return groups;
+    }
+
+    let state = null;
+
+    for (const directive of directives) {
+        if (state) {
+            if (directive.unprocessedDirective.parentComment === state.parentComment) {
+                state.group.push(directive);
+            } else {
+                groups.push(state.group);
+                state = null;
+            }
+        }
+
+        state = {
+            group: [directive],
+            parentComment: directive.parentComment
+        };
+    }
+
+    if (state && state.group.length) {
+        groups.push(state.group);
+    }
+
+    return groups;
+}
+
+/**
+ * Creates removal details for a set of directives within the same comment.
+ * @param {Directive[]} directives Unused directives to be removed.
+ * @param {Token} commentToken The backing Comment token.
+ * @param {SourceCode} sourceCode A SourceCode object for the given text.
+ * @returns {{ fix, position, ruleIds }[]} Details for later creation of output Problems.
+ */
+function createIndividualDirectivesRemoval(directives, commentToken, sourceCode) {
+    const commentStart = sourceCode.lineStartIndices[commentToken.loc.start.line - 1] + commentToken.loc.start.column;
+    const commentContents = commentToken.value.split("--")[0].trimRight();
+    let lastRuleEnd = 0;
+
+    return directives.map(directive => {
+        const ruleStart = commentContents.indexOf(directive.ruleId, lastRuleEnd);
+        const ruleEnd = lastRuleEnd = ruleStart + directive.ruleId.length;
+
+        return {
+            description: directive.ruleId,
+            fix: {
+                range: [
+                    commentStart + 2 + ruleStart,
+                    commentStart + 2 + ruleEnd + (commentContents[ruleEnd] === "," ? 1 : 0)
+                ],
+                text: ""
+            },
+            position: directive.unprocessedDirective
+        };
+    });
+}
```

Comment:
This could still find a wrong index, for example:

```js
/* eslint no-undef: 2, no-undefined: 2 */

/* eslint-disable-next-line no-undefined, no-undef */
undefined;

```

will be autofixed to:

```js
/* eslint no-undef: 2, no-undefined: 2 */

/* eslint-disable-next-line ined,  */
undefined;

```

(first pass removed `no-undef` from `no-undefined`, the second removed `no-undef`).

Could we maybe use regular expressions to more precisely locate the rule? It could be something like this:

```suggestion
function createIndividualDirectivesRemoval(directives, commentToken, sourceCode) {
    const listOffset = /^\s*\S+\s+/u.exec(commentToken.value)[0].length;
    const listText = commentToken.value
        .slice(listOffset) // remove eslint-*
        .split(/\s-{2,}\s/u)[0] // remove -- directive comment
        .trimRight();
    const listStart = commentToken.range[0] + 2 + listOffset;

    return directives.map(directive => {
        const { ruleId } = directive;
        const match = new RegExp(String.raw`(?:^|,)\s*${ruleId}\s*(?:$|,)`, "u").exec(listText);
        const ruleOffset = match.index;
        const ruleEndOffset = ruleOffset + match[0].length;
        const ruleText = listText.slice(ruleOffset, ruleEndOffset);

        return {
            description: ruleId,
            fix: {
                range: [
                    listStart + ruleOffset + (ruleText.startsWith(",") && ruleText.endsWith(",") ? 1 : 0),
                    listStart + ruleEndOffset
                ],
                text: ""
            },
            position: directive.unprocessedDirective
        };
    });
}
```